internal/resources/smbshare.go

```diff
@@ -339,6 +348,14 @@ func (m *SmbShareManager) updateClusteredState(
 		return Requeue
 	}
 
+	changed, err := m.claimOwnership(ctx, planner.SmbShare, statefulSet)
+	if err != nil {
+		return Result{err: err}
+	} else if changed {
```

Comment:
Prefer `if changed` (no need to have `else if`).

Reply:
There are a lot of:
```
if err != nil {
   return ...
} else if abc {
   return ...
}
```

pattern in the current smbshare.go. I am going to leave them alone for now but we can do a follow up pr that just does a style change to *all* of these in one batch. Sound ok?

Reply:
Sure. I will take a closer look on this pattern cross all code-base.

Reply:
Resolving this conversation as we've discussed we'll do a general if-else-if with returns style clean-up in a follow up PR.